From foo@baz Tue Jun  2 22:45:46 PDT 2009
Date: Tue, 02 Jun 2009 22:45:46 -0700
To: Greg KH <greg@kroah.com>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: Staging: pata_rdc: convert code to work in 2.6.29

This fixes build problems in the pata_rdc driver due to api changes in
the libata layer.


Cc: Kevin Huang  <Kevin.Huang@rdc.com.tw>
Cc: Tomy Wang <Tomy.Wang@rdc.com.tw>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/Kconfig             |    2 ++
 drivers/staging/Makefile            |    2 +-
 drivers/staging/pata_rdc/Kconfig    |    6 ++++++
 drivers/staging/pata_rdc/Makefile   |    2 ++
 drivers/staging/pata_rdc/pata_rdc.c |    4 ++--
 5 files changed, 13 insertions(+), 3 deletions(-)

--- a/drivers/staging/pata_rdc/pata_rdc.c
+++ b/drivers/staging/pata_rdc/pata_rdc.c
@@ -148,7 +148,7 @@ static int __devinit rdc_init_one(
 {
     //struct device *dev = &pdev->dev;
     struct ata_port_info port_info[2];
-    struct ata_port_info *ppinfo[] = { &port_info[0], &port_info[1] };
+    const struct ata_port_info *ppinfo[] = { &port_info[0], &port_info[1] };
 
     int rc;
 
@@ -174,7 +174,7 @@ static int __devinit rdc_init_one(
 
     pci_intx(pdev, 1);  // enable interrupt
 
-    return ata_pci_init_one(pdev, ppinfo);
+    return ata_pci_sff_init_one(pdev, ppinfo, &rdc_pata_sht, NULL);
 }
 
 // callback function for ata_port
